src/compiler/crystal/macros/methods.cr

```diff
@@ -142,6 +177,20 @@ module Crystal
       raise SkipMacroException.new(@str.to_s)
     end
 
+    def interpret_read_file(node)
```

Comment:
Indeed, after all I reckon that sharing logic between `require` / `run` and `read_file` is not desirable. Former ones use it to require/run code, latter is used to read generic files, not necessarily code, so having such behaviour would be confusing.

Reply:
Seems like GTG if that's the last remaining thing :shipit:

Reply:
But then the `find_file` refactor is not required to add the feature. Better to have a different PR for that refactor. Otherwise tracking the change on releases is harder ;-)

Reply:
@bcardiff Fair 'nuff, I've reverted this.